src/v2/ion/uiSchemaTransformer.js

```diff
@@ -61,6 +77,7 @@ const createUISchema = (remediationValue = []) => {
     if (ionFormField.name === 'factorId' ||
       ionFormField.name === 'factorProfileId') {
       uiSchema.type = 'factorType';
+      addFactorTypeToOptions(ionFormField.options, factors);
```

Comment:
this is a bit tricky since looks like you change the (original) `options` in place hence it works.
I'd suggest to 
`uiSchema.options = createFactorTypeOptions(ionFormField.options, factors);`